src/Blockcore.Indexer.Core/Storage/Mongo/MongoStorageOperations.cs

```diff
@@ -222,30 +229,18 @@ public SyncBlockInfo PushStorageBatch(StorageBatch storageBatch)
          // allow any extensions to push to repo before we complete the block.
          OnPushStorageBatch(storageBatch);
 
-         string lastBlockHash = null;
-         long blockIndex = 0;
-         var markBlocksAsComplete = new List<UpdateOneModel<BlockTable>>();
-         foreach (BlockTable mapBlock in storageBatch.BlockTable.Values.OrderBy(b => b.BlockIndex))
-         {
-            FilterDefinition<BlockTable> filter =
-               Builders<BlockTable>.Filter.Eq(block => block.BlockIndex, mapBlock.BlockIndex);
-            UpdateDefinition<BlockTable> update =
-               Builders<BlockTable>.Update.Set(blockInfo => blockInfo.SyncComplete, true);
-
-            markBlocksAsComplete.Add(new UpdateOneModel<BlockTable>(filter, update));
-            lastBlockHash = mapBlock.BlockHash;
-            blockIndex = mapBlock.BlockIndex;
-         }
-
          // mark each block is complete
-         data.BlockTable.BulkWrite(markBlocksAsComplete, new BulkWriteOptions() { IsOrdered = true });
```

Comment:
ah this we have to pay attention here, notice `IsOrdered = true` this means we respect the update blocks in the order of the list, this is important in case of failure we must maintain that the blocks update in order becuase then when the node will restart we look for the last uncomplete block (not last) and we may miss an uncomplete block.

However I am not sure  that this can fail partially because we send a bulk update to mongo but we need to be aware of this

Reply:
@DavidGershony this is the last issue on this PR.